applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp

```diff
@@ -16,6 +16,7 @@
 
 #include "custom_utilities/qsvms_data.h"
 #include "custom_utilities/time_integrated_qsvms_data.h"
+#include "../SwimmingDEMApplication/custom_utilities/qsvmsdemcoupled_data.h"
```

Comment:
Considering that the fluid application is considered a core application, I don't think it must have any include from any other application. This is even worse in this case, as this is the base element source file.

Wouldn't it work if you do the explicit instantiation in the cpp of your element?

Reply:
yep this is a blocker for me for this reason

I am quite convinced that it should work if you add this in your element

Reply:
I have tried adding the explicit instantiations in the cpp of the new element and removing them and the includes in the elements from the _FluidDynamicsApplication_ but I got links errors when I compiled Kratos. Adding the includes in the fluid was the alternative to solve the errors.

![Selección_098](https://user-images.githubusercontent.com/39963156/94029491-26cf7d80-fdbd-11ea-89ca-1e4b073efeaf.png)

Reply:
I guess that you already tried to put the explicit instantiation of the base fluid element before the qs one (which is before the qsdem one).

Reply:
Are you guys sure that it is possible to instantiate explicitely a class from a different library?
I mean, doing this:
https://github.com/KratosMultiphysics/Kratos/blob/3268ba4560cd005ece94fe4b97e7f7b48f92cd86/applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp#L893
From the SwimmingDEM. If the template was in the header file, it would be easy, but being in a cpp, you instantiate them all in the fluid cpp and close the door to extra instantiations from other apps, no?

Reply:
Maybe including the cpp file from the SwimmingDEM by #include "fluid_element.cpp"?

Reply:
Thanks to changes proposed by @maceligueta, I solved the link errors. I've removed the changes in the _FluidDynamicsApplication_ and Kratos compiles.

Reply:
> 
> 
> Maybe including the cpp file from the SwimmingDEM by #include "fluid_element.cpp"?

This is definitively the way to do it. Thanks @maceligueta !